docs/mds_test_plan.md

```diff
@@ -0,0 +1,39 @@
+# Metadata Service (mds) test plan
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [[no-file-name-irregular-characters] Do not use `_` in a file name](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)